lib/flutter/flutter_launch.dart

```diff
@@ -195,55 +179,91 @@ class _RunLaunchInstance extends _LaunchInstance {
       project,
       killHandler: _kill,
       cwd: project.path,
-      title: description,
+      title: 'flutter run ${_target} ($_mode)',
       targetName: _device?.name
     );
-    launchManager.addLaunch(_launch);
   }
 
   bool get pipeStdio => false;
 
   Future<Launch> launch() async {
-    FlutterTool flutter = _flutterSdk.sdk.flutterTool;
-
-    _runner = _flutter(flutter, _args, cwd: project.path);
-    _runner.execStreaming();
-    _runner.onStdout.listen((String str) {
-      _watchForObservatoryMessage(str);
-      _launch.pipeStdio(str);
-    });
-    _runner.onStderr.listen((String str) => _launch.pipeStdio(str, error: true));
-    _runner.onExit.then((code) => _launch.launchTerminated(code));
+    return daemon.app.start(
+      _device?.id,
+      project.path,
+      mode: _mode.name,
+      startPaused: _mode.startPaused,
+      target: _target,
+      route: _route
+    ).then((AppStartedResult result) {
+      _app = daemon.app.createDaemonApp(result.appId, supportsRestart: result.supportsRestart);
+      _launch.app = _app;
+
+      launchManager.addLaunch(_launch);
+
+      _LogStatusJob job;
+
+      _app.onDebugPort.then((DebugPortAppEvent event) {
+        _observatoryPort = event.port;
+        new Future.delayed(new Duration(milliseconds: 100), _connectToDebugger);
```

Comment:
What happens if you don't wait 100ms ? Add a comment indicating the problem that this delay addresses?